### ISSUE TITLE
Segfault when using const value as argument for macro function.

### ISSUE THREAD

**Issue by tobilinz** [oven-sh/bun#6014]
### What version of Bun is running?

1.0.3

### What platform is your computer?

Linux 6.5.3-arch1-1 x86_64 unknown

### What steps can reproduce the bug?

1. Create a new bun project using `bun init -y`
2. Paste the following code into `index.ts`:
```
import { myMacro } from './other' with { type: 'macro' };
const test = 'Hello World ';
console.log(myMacro(test));
```
5. Paste the following code into a new file called `other.ts`:
```
export function myMacro(value: string): string {
  return value + "and Bun!";
}
```
7. Run `bun build ./index.ts`

### What is the expected behavior?

Console output:
```
// index.ts
console.log("Hello World and Bun!");
```

### What do you see instead?

Console output:
`[1]    27447 segmentation fault (core dumped)  bun build ./index.ts`

### Additional information

This bug also occurs when importing a const from another file  or importing  the content of a .txt* file and using it as input for a macro function.

* .wgsl files also cause this behavior. Other file types have not been tested.

**Comment by tobilinz** [comment]
I accidentally closed it.

**Comment by maxmilton** [comment]
Possibly related: https://github.com/oven-sh/bun/issues/3832 and https://github.com/oven-sh/bun/issues/3830#issuecomment-1664829699

**Comment by Electroid** [comment]
Closed by #6756

**Comment by maxmilton** [comment]
As mentioned in https://github.com/oven-sh/bun/issues/3830 the segfault is now correctly handled, however there's still an issue with handling certain values in macros. Not sure this should be closed yet.
